libp2p/multicodec.nim

```diff
@@ -251,37 +257,32 @@ const
 
 proc multiCodec*(name: string): MultiCodec {.compileTime.} =
   ## Generate MultiCodec from string ``name`` at compile time.
-  var code = NameCodecs.getOrDefault(name, -1)
-  if code == -1:
-    raise newException(MultiCodecError,
-                       "MultiCodec `" & name & "` not supported!")
-  result = MultiCodec(code)
+  let code = NameCodecs.getOrDefault(name, -1)
+  doAssert(code != -1)
+  MultiCodec(code)
 
 proc multiCodec*(code: int): MultiCodec {.compileTime.} =
   ## Generate MultiCodec from integer ``code`` at compile time.
-  var name = CodeCodecs.getOrDefault(code, "")
-  if name == "":
-    raise newException(MultiCodecError,
-                       "MultiCodec with code " & $code & " not supported!")
-  result = MultiCodec(code)
+  let name = CodeCodecs.getOrDefault(code, "")
+  doAssert(name != "")
```

Comment:
This is COMPILE TIME code, asserting is fine as gives complete __LINE__ and __FILE__

Reply:
need a `static_assert` I think ;) until then, let's maybe put a comment, I certainly missed that it's a compile-time function (I kind of don't like the `compileTime` pragma, it's such an arbitrary distinction)

Reply:
`static: doAssert` will work I think